src/dbnode/persist/fs/retriever_test.go

```diff
@@ -131,74 +134,124 @@ func TestBlockRetrieverHighConcurrentSeeksCacheShardIndices(t *testing.T) {
 }
 
 func testBlockRetrieverHighConcurrentSeeks(t *testing.T, shouldCacheShardIndices bool) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
 	defer leaktest.CheckTimeout(t, 2*time.Minute)()
 
 	dir, err := ioutil.TempDir("", "testdb")
 	require.NoError(t, err)
 	defer os.RemoveAll(dir)
-	filePathPrefix := filepath.Join(dir, "")
-	fsOpts := testDefaultOpts.SetFilePathPrefix(filePathPrefix)
 
-	fetchConcurrency := 4
-	seekConcurrency := 4 * fetchConcurrency
-	opts := testBlockRetrieverOptions{
-		retrieverOpts: defaultTestBlockRetrieverOptions.
-			SetFetchConcurrency(fetchConcurrency).
-			SetRequestPoolOptions(pool.NewObjectPoolOptions().
-				// NB(r): Try to make sure same req structs are reused frequently
-				// to surface any race issues that might occur with pooling.
-				SetSize(fetchConcurrency / 2)),
-		fsOpts: fsOpts,
-	}
+	// Setup retriever.
+	var (
+		filePathPrefix = filepath.Join(dir, "")
+		fsOpts         = testDefaultOpts.SetFilePathPrefix(filePathPrefix)
+
+		fetchConcurrency           = 4
+		seekConcurrency            = 4 * fetchConcurrency
+		updateOpenLeaseConcurrency = 4
+		opts                       = testBlockRetrieverOptions{
+			retrieverOpts: defaultTestBlockRetrieverOptions.
+				SetFetchConcurrency(fetchConcurrency).
+				SetRequestPoolOptions(pool.NewObjectPoolOptions().
+					// NB(r): Try to make sure same req structs are reused frequently
+					// to surface any race issues that might occur with pooling.
+					SetSize(fetchConcurrency / 2)),
+			fsOpts: fsOpts,
+		}
+	)
 	retriever, cleanup := newOpenTestBlockRetriever(t, opts)
 	defer cleanup()
 
-	nsMeta := testNs1Metadata(t)
-	ropts := nsMeta.Options().RetentionOptions()
-	nsCtx := namespace.NewContextFrom(nsMeta)
+	// Setup the open seeker function to fail sometimes to exercise that code path.
+	seekerMgr := retriever.seekerMgr.(*seekerManager)
+	existingNewSeekerFn := seekerMgr.newOpenSeekerFn
+	newNewSeekerFn := func(shard uint32, blockStart time.Time, volume int) (DataFileSetSeeker, error) {
```

Comment:
nit: Is this meant to be `newOpenSeekerFn` instead of `newNewSeekerFn`?

Reply:
Yeah I can make this less confusing